### DISCUSSION TITLE
Why can't JUnit accept Console input from the user?

### DISCUSSION THREAD

**Discussion by havefunatcode** [junit-team/junit-framework#3766] [Q&A]
I understand that JUnit itself is all about unit testing, so it's designed to not accept input from the user. 
I'd like to see the code that is structured to prevent me from getting console input, but where should I look? 
Most references will tell you that you can use System.setIn() to test against standard input. 
I don't want to test against standard input, I want to see where JUnit doesn't accept console input. 
I also want to know how `public static void main(String[] args)` accepts Console input.

To verify this, I compiled the following code and checked the Bytecode.
```
public static void main(String[] args) {
    Scanner scanner = new Scanner(System.in);
    System.out.println("Please write your input String");
    String inputStr = scanner.nextLine();
    System.out.println(inputStr);
}

@Test
void scannerTest() {
    Scanner scanner = new Scanner(System.in);
    System.out.println("Please write your input String");
    String inputStr = scanner.nextLine();
    System.out.println(inputStr);
}
```
However, I couldn't find where it was blocking input to the console.

**Comment by sbrannen** [comment]
> However, I couldn't find where it was blocking input to the console.

JUnit doesn't block console input.

I just ran your test and typed in "foo" and then ENTER, and this is the result:

```
Please write your input String
foo
foo
```

You just need to figure out where the "console" is when you're running JUnit.

In Eclipse IDE (where I ran this), there is a `Console` window.
